src/main/java/edu/harvard/iq/dataverse/license/LicenseListingPage.java

```diff
@@ -55,4 +58,26 @@ public String init() {
         return StringUtils.EMPTY;
     }
 
+    public void saveLicenseActiveStatus(LicenseDto licenseDto) {
+
+        if (licenses.stream().noneMatch(LicenseDto::isActive)) {
+            licenseDto.setActive(true);
+            displayNoLicensesActiveWarningMessage();
+            return;
+        }
+
+        License license = licenseDAO.find(licenseDto.getId());
+        license.setActive(licenseDto.isActive());
+        licenseDAO.saveChanges(license);
+    }
+
+    // -------------------- PRIVATE --------------------
+
+    private void displayNoLicensesActiveWarningMessage() {
+        FacesContext.getCurrentInstance().
+                addMessage(null, new FacesMessage(FacesMessage.SEVERITY_ERROR,
+                        StringUtils.EMPTY,
+                        BundleUtil.getStringFromBundle("dashboard.license.warning")));
```

Comment:
Use JsfHelper class for this

Reply:
It is also tied with problem that passing parameter to template to display errors wont work, and using JsfHelper doesn't work with this method.

Reply:
Ah now I know why, JsfHelper is using flash scope which display error message only after redirection attempt which we are not doing right here.